content/departments/product-engineering/engineering/enablement/dev-experience/newsletter.md

```diff
@@ -8,6 +8,60 @@ To learn more about components of Sourcegraph's developer experience, check out
 
 > NOTE: For authors, refer to [this guide](./index.md#newsletter) for preparing a newsletter.
 
+## Jan 10, 2022
+
+Happy new year, and welcome to another iteration of the [Developer Experience newsletter](./newsletter.md)!
+
+To have your updates highlighted here, please tag your PR or issue with the `dx-announce` label! If you have questions or feedback, feel free to reach out in #dev-experience or in our [discussions](https://github.com/sourcegraph/sourcegraph/discussions/categories/developer-experience) as well.
+
+### Backward-compatible database migrations are now enforced
+
+Backward-compatible database migrations are now enforced in the CI pipeline for `sourcegraph/sourcegraph` - see the PR to re-enable the check at [#28872](https://github.com/sourcegraph/sourcegraph/pull/28872). This PR contains some initial documentation on writing backwards-compatible migrations, but it is still a work in progress.
+
+**What is a backwards compatible migration?:** A migration is backwards-compatible with a particular Sourcegraph version if those changes can be applied to a version without ill-effect.
+
+**What has already changed? (TL;DR)**: We've removed our use of golang-migrate that ran database migrations on startup of the frontend service and added a `migrator` service that runs database migrations separately from and prior to instance upgrades. This puts us well on our way to removing the entire class of frequent "dirty database" bugs that plagues many site-administrators on every upgrade.
+
+**What else is changing?:** We will soon be enforcing that the unit tests of the _previous minor release_ continue to pass with the newest database schema. This gives high confidence that any changes to the database will not negatively affect a running instance (behind at most one minor version). This allow site-administrators to upgrade an instance without requiring downtime to run the migrations.
+
+Of course, this check will come with escape hatches in the event of flake or test failures that are locked in the past. We're currently fleshing out the documentation on the subject, so keep an eye out for updates!
+
+For the full announcement or to leave comments, check out [the Slack discussion](https://sourcegraph.slack.com/archives/C0EPTDE9L/p1641329708282700)!
```

Comment:
cc @efritz this is a condensed version of your announcement, let me know if it does the job!

Reply:
Excellent!